### PR TITLE
feat: argument_input_types

### PR DESCRIPTION
**Pull request by barnabasJ**
### Contributor checklist

- [ ] Bug fixes include regression tests
- [x] Features include unit/acceptance tests

### CODE REVIEW

**Review by zachdaniel — DISMISSED** [review]

**Comment by zachdaniel** [comment]
LGTM, feel free to merge when the build passes 🥳

**Comment by barnabasJ** [comment]
I fixed the CI issues, can't merge without another review

**Comment by zachdaniel** [comment]
🚀 Thank you for your contribution! 🚀

**Pull request merged**
